controllers/internal/utils/dns.go

```diff
@@ -20,30 +20,99 @@ package utils
 import (
 	"fmt"
 	"sort"
+	"strings"
 
 	"github.com/lixiangzhong/dnsutil"
+	"github.com/miekg/dns"
 )
 
+type DNSServer struct {
+	Host string
+	Port int
+}
+
 // Dig retrieves list of tuple <IP address, A record > from edge DNS server for specific FQDN
 func Dig(edgeDNSServer, fqdn string) ([]string, error) {
```

Comment:
the old `Dig` is still being called from couple of tests so I was thinking to left it there, because it's easier to call it as a one-liner rather than preparing that array first (4 lines or so), but the var args function will probably do the trick, so yea, let me refactor it 👍  then it would be called w/ the `...` expansion on that array, right?

Reply:
Should we keep the function just because it is used by tests ?

Reply:
hmm, you are right, I will change that to one single `Dig` implementation that will do the fallback